aptos/api/aux-ts/src/graphql/resolvers/query.ts

```diff
@@ -328,7 +332,72 @@ export const query = {
       hasAuxAccount: auxAccount !== undefined,
     };
   },
+  async stakePool(
+    _parent: any,
+    { stakePoolInput }: QueryStakePoolArgs
+  ): Promise<Maybe<StakePool>> {
+    const coinInfoReward = await auxClient.getCoinInfo(
+      stakePoolInput.coinTypeReward
+    );
+    const coinInfoStake = await auxClient.getCoinInfo(
+      stakePoolInput.coinTypeStake
+    );
+    const poolClient = await new StakePoolClient(auxClient, {
+      coinInfoReward,
+      coinInfoStake,
+    });
+    const pool = await poolClient.query();
+
+    // @ts-ignore
```

Comment:
Agree this is confusing. It seems like we might be able to return `Partial` types and avoid `ts-ignore` all together? Maybe @lidatong can confirm, i think this has to do with how resolvers work.

Reply:
Hackery unrelated to the return values, just other scattered `ts-ignores` that do some ad-hoc object manipulation on pools.

> typescript just doesn't know that apollo server is linking the function resolvers with the literal fields on the object.

`Partial` should resolve this though right?

Reply:
Sure you're right that partial is better than ts ignore since it'll catch some errors like field typos. It won't catch forgetting a field though

[Pick](https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys) is probably the most disciplined thing to use but I don't think it's necessary